### DISCUSSION TITLE
How to use nuxt composables in a module?

### DISCUSSION THREAD

**Discussion by gzf6** [nuxt/nuxt#25369] [Questions]
When I import the module I developed into other modules, I get the error
`Could not resolve "#imports"`
`You can mark the path "#imports" as external to exclude it from the bundle, which will remove this  error and leave the unresolved path in the bundle.`

**Comment by huang-julien** [comment]
Hello :wave: composables are meant to be ran in a vue context which is something runtime

**Reply by gzf6** [reply]
If I want to use useAsyncData, I have to manually import it from nuxt, right?

**Reply by huang-julien** [reply]
no you can't use `useAsyncData` within a module.

A module is build-time only, meaning that no vue nor nuxt app context is available. 

What is your use case ? you probably don't need `useAsyncData`

**Reply by gzf6** [reply]
Just simply extracted business components. Use useAsyncData when getting data in the template. As you said, I should use $fetch right?

**Reply by huang-julien** [reply]
I'm not sure to quite understand.

When you say "module", are you talking about a nuxt module ?

Do you have a small reproduction of your issue ?

**Reply by TheAlexLichter** [reply]
If I understand correctly, @gzf6 tries to encapsulate some composables in an own module - which is fine.

But AFAIK auto-imports are not supported in modules and you have to import via `#imports`

**Comment by gzf6** [comment]
> If I understand correctly, @gzf6 tries to encapsulate some composables in an own module - which is fine.
> 
> But AFAIK auto-imports are not supported in modules and you have to import via `#imports`

@manniL Thank you very much for explaining.
@huang-julien  That's what I mean.

**Comment by szulcus** [comment]
In [documentation ](https://nuxt.com/docs/guide/going-further/modules#runtime-directory)I read:
"(...) If you are using the module starter, auto-imports will not be enabled in your playground either."

But in nuxt module starter, in playground I can use autoimports without any warnings. I see error only in real project. How I can enable these errors in playground?
